### ISSUE TITLE
Support custom url schemes / AppExtension

### ISSUE THREAD

**Issue by jberkel** [kiwix/iOS#15]
The use case is the following: a 3rd party app wants to display some of the content stored in kiwix, for example a specific Wiktionary entry. 

It would be great if this content could be address directly, either with a custom url scheme or as an app extension.

**Comment by kelson42** [comment]
I support this feature request. This is possible in Android AFAIK with our custom zim:// url scheme.

**Comment by automactic** [comment]
In 1.7 and up, you can open URL like 

- kiwix://zimID/contentPath

Will support but not yet supported:

- kiwix://search/searchTerm
- kiwix://zimID/articleName
- kiwix://zimName(PID)/contentPath
- kiwix://bookmark/articleName

**Comment by jberkel** [comment]
@automactic thanks! what's the format of contentPath?

**Comment by automactic** [comment]
@jberkel The relative path of a content. For example, in `wikipedia_en_simple_all_2016-08.zim`, the article "Basic English" has the url `kiwix://2c81e7cf-f929-5bb3-76ec-ee72f5949e83/A/Basic_English.html`

**Comment by jberkel** [comment]
Yes. In my specific case it'd like to link to an entry on Wiktionary. I already have the page name, so would expect something like `kiwix://wiktionary_en/Page` to work.

**Comment by automactic** [comment]
@jberkel  Yes, the zimName(PID) is a human readable ID that stay stable between different versions of the zim file. Some of the zim has it, mostly don't, but eventually all of them will have it. Kiwix for iOS cannot read the string out of zim files now, but in 1.8 (in development) it can. 

Plus, there are a lot more to do before we can release this function. For example, a custom 404 providing recovery options.

Anyway, we will get there eventually. And I will be happy to invite you to join our TestFlight once this feature is available. ☺️

**Comment by jberkel** [comment]
@automactic what's the state of this? perhaps this  can be closed now?

**Comment by jberkel** [comment]
How do I get the ID from an installed ZIM file? There's an ID field in the app but it doesn't look like the IDs/uuids mentioned earlier in this thread.

![IMG_0152](https://user-images.githubusercontent.com/18891/65594235-a6e8ba80-df92-11e9-9e61-a90b3a6e88a6.PNG)

**Comment by kutulus** [comment]
> In 1.7 and up, you can open URL like
> 
>     * kiwix://zimID/contentPath
> 
> 
> Will support but not yet supported:
> 
>     * kiwix://search/searchTerm
>     * kiwix://zimID/articleName
>     * kiwix://zimName(PID)/contentPath
>     * kiwix://zimFileName/contentPath
>     * kiwix://bookmark/articleName

hi,
are there any plans for a realisation?

zim://zimFileName/contentPath

It would be extremely helpful as it would allow updates of zim files without breaking links in external apps.
Interlinks between zim archives would also be easier to realis
